### PR TITLE
fix: add secret only when not in secret file

### PR DESCRIPTION
**Pull request by nickmansrob**
This PR fixes issue #2716 by adding logic to make sure a ENV variable which is specified in a Secret is not added as an empty ENV variable too, resulting in a duplicate key (one from the Secret, one as the default value in OperatorConfiguration which is "").

### CODE REVIEW

File: pkg/cluster/k8sres.go [hunk 1]
```diff
@@ -2481,7 +2481,9 @@ func (c *Cluster) generateLogicalBackupPodEnvVars() []v1.EnvVar {
 		}
 
 	case "gcs":
-		envVars = append(envVars, v1.EnvVar{Name: "LOGICAL_BACKUP_GOOGLE_APPLICATION_CREDENTIALS", Value: c.OpConfig.LogicalBackup.LogicalBackupGoogleApplicationCredentials})
+        if c.OpConfig.LogicalBackup.LogicalBackupGoogleApplicationCredentials != "" {
+          envVars = append(envVars, v1.EnvVar{Name: "LOGICAL_BACKUP_GOOGLE_APPLICATION_CREDENTIALS", Value: c.OpConfig.LogicalBackup.LogicalBackupGoogleApplicationCredentials})
+        }
```
Comment (FxKu) [inline pkg/cluster/k8sres.go:2486]:
```suggestion
		if c.OpConfig.LogicalBackup.LogicalBackupGoogleApplicationCredentials != "" {
			envVars = append(envVars, v1.EnvVar{Name: "LOGICAL_BACKUP_GOOGLE_APPLICATION_CREDENTIALS", Value: c.OpConfig.LogicalBackup.LogicalBackupGoogleApplicationCredentials})
		}
```

File: pkg/cluster/k8sres.go [hunk 2]
```diff
@@ -2492,11 +2494,11 @@ func (c *Cluster) generateLogicalBackupPodEnvVars() []v1.EnvVar {
...
 				Value: c.OpConfig.LogicalBackup.LogicalBackupAzureStorageContainer,
-			},
-			{
-				Name:  "LOGICAL_BACKUP_AZURE_STORAGE_ACCOUNT_KEY",
-				Value: c.OpConfig.LogicalBackup.LogicalBackupAzureStorageAccountKey,
 			}}...)
+
+		if c.OpConfig.LogicalBackup.LogicalBackupAzureStorageAccountKey != "" {
+			envVars = append(envVars, v1.EnvVar{Name: "LOGICAL_BACKUP_AZURE_STORAGE_ACCOUNT_KEY", Value: c.OpConfig.LogicalBackup.LogicalBackupAzureStorageAccountKey})
+		}
```
Comment (FxKu) [inline pkg/cluster/k8sres.go:2501]:
```suggestion
		if c.OpConfig.LogicalBackup.LogicalBackupAzureStorageAccountKey != "" {
			envVars = append(envVars, v1.EnvVar{Name: "LOGICAL_BACKUP_AZURE_STORAGE_ACCOUNT_KEY", Value: c.OpConfig.LogicalBackup.LogicalBackupAzureStorageAccountKey})
		}
```

**Comment by FxKu** [comment]
:+1:

File: pkg/cluster/k8sres.go (hunk 1 above)
Comment (nickmansrob) [reply pkg/cluster/k8sres.go]: Seems my formatter did not catch that one :thinking:

**Pull request merged**